### DISCUSSION TITLE
[Native Filters] Support Instant Filtering like legacy filter charts

### DISCUSSION THREAD

**Discussion by kamalkeshavani-aiinside** [apache/superset#20663] [Ideas]
One big concern before moving from legacy filters to native filters for our team is that it will increase user clicks.
We mostly used 'Instant filtering' option of legacy filters, since we usually have just 1-2 filters in our dashboard. If we move to native filters, user need to click on 'Apply' every time after changing the filter.
![image](https://user-images.githubusercontent.com/74634977/178167577-999994dc-9597-4c78-b404-c96fc8da19cd.png)

Since legacy filters are going to be deprecated soon, it would be great if we can bring this feature to native filters.

**Comment by sfirke** [comment]
+1 -- I'm coming to Superset from a different BI platform that had the option of instant filtering and it was a nice boost to user-friendliness on a dashboard with just one filter.  Would be great to see that in Superset, too.

**Comment by ndeebook** [comment]
`Superset is a modern data exploration and data visualization platform`
This is a must have for the `exploration` part. I'm surprised there is not more people asking for this.

**Comment by mdeshmu** [comment]
Seems like it was there in original implementation of native filters but removed later due to performance implication https://github.com/apache/superset/pull/15365
Probably this can be added back behind a feature flag. cc: @villebro for awareness.

**Comment by xavier-GitHub76** [comment]
Hello,

this feature will be implemented ?

Best regards

**Comment by maximStafeevAtAllscripts** [comment]
Hi, I'm also interested in this feature!

**Comment by keiwhy** [comment]
I would like this feature to be implemented. Instant filtering option was the first thing the user of the dashboards commented on. Clicking "Apply filter"-button is for them slow and cumbersome.  Adding "auto_apply": true into metadata does not seem to work.

**Comment by kuvsh** [comment]
Hello! Do you have any updates regarding this topic? We are really missing this option to improve our customers user experience.

**Comment by davidriverapericas** [comment]
+1, it would be really nice to have this feature
